github_connector/models/github_repository.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from openerp import models, fields, api
+
+_logger = logging.getLogger(__name__)
+
+
+class GithubRepository(models.Model):
+    _name = 'github.repository'
+    _inherit = ['abstract.github.model']
+    _order = 'organization_id, name'
+
+    _github_type = 'repository'
+    _github_login_field = 'full_name'
+
+    # Column Section
+    organization_id = fields.Many2one(
+        comodel_name='github.organization', string='Organization',
+        required=True, index=True, readonly=True, ondelete='cascade')
+
+    name = fields.Char(
+        string='Name', index=True, required=True, readonly=True)
+
+    complete_name = fields.Char(
+        string='Complete Name', readonly=True,
+        compute='_compute_complete_name', store=True)
+
+    description = fields.Char(string='Description', readonly=True)
+
+    website = fields.Char(string='Website', readonly=True)
+
+    repository_branch_ids = fields.One2many(
+        comodel_name='github.repository.branch',
+        inverse_name='repository_id', string='Branches', readonly=True)
+
+    repository_branch_qty = fields.Integer(
+        string='Branches Quantity', compute='_compute_repository_branch_qty',
```

Comment:
Better "Number of" everywhere (http://www.grammar-monster.com/easily_confused/amount_quantity_and_number.htm most important is that Quantity doesnot suit for animated objects)